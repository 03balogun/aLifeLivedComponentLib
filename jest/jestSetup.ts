/* eslint-disable no-undef */
jest.useFakeTimers('legacy'); // Stops reference error from appearing
