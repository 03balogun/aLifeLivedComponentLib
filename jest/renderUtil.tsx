import React from 'react';
import { ThemeProvider } from '../src/themes/ThemeProvider';
import { render as rtlRender } from '@testing-library/react-native';

function render(ui, { ...renderOptions } = {}) {
  function Wrapper({ children }) {
    return <ThemeProvider>{children}</ThemeProvider>;
  }

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything for convenience
export * from '@testing-library/react-native';

// override render method
export { render };
