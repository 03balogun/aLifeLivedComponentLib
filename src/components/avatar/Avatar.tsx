import React from 'react';
import FastImage, { Priority, ResizeMode } from 'react-native-fast-image';
import { Box } from '../../themes/box/Box';
import {
  StyledImageProps,
  SystemImage,
} from '../../themes/systemImage/SystemImage';

type AvatarProps = {
  resizeMode?: ResizeMode;
  source: {
    uri: string;
    priority: Priority;
  };
};

export const Avatar: React.FC<AvatarProps & StyledImageProps> = ({
  source,
  resizeMode = 'cover',
  variant = 'roundAvatar',
  ...rest
}) => {
  return !source.uri ? (
    <Box
      borderWidth={0}
      padding="m"
      borderRadius="xXXlarge"
      alignItems="center"
      justifyContent="center"
      backgroundColor="primary"
      {...rest}
    />
  ) : (
    <SystemImage
      source={{
        uri: source.uri,
        priority: FastImage.priority[source.priority],
      }}
      resizeMode={FastImage.resizeMode[resizeMode]}
      variant={variant}
      {...rest}
    />
  );
};
