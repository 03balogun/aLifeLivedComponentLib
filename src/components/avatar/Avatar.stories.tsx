import React from 'react';

import { storiesOf } from '@storybook/react-native';
// Component
import { Avatar } from './Avatar';
import { CenterView } from '../centerView/CenterView';

storiesOf('Avatar', module)
  .addDecorator(Story => <CenterView>{Story()}</CenterView>)
  .add('Default avatar', () => (
    <Avatar source={{ uri: 'test' }} variant="default" />
  ));
