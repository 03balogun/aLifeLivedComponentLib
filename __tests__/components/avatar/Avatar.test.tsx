import React from 'react';
import { Priority } from 'react-native-fast-image';
import { render } from './../../../jest/renderUtil';

// Component
import { Avatar } from '../../../src/components/avatar/Avatar';

const setup = (
  source: {
    uri: string;
    priority: Priority;
  },
  variant: 'roundAvatar'
) => render(<Avatar source={source} resizeMode="center" variant={variant} />);

describe('components >> avatar ', () => {
  describe('Round avatar', () => {
    test('Should render round avatar correctly', () => {
      const { toJSON } = setup(
        { uri: 'https://picsum.photos/id/237/200/300', priority: 'normal' },
        'roundAvatar'
      );
      expect(toJSON()).toMatchSnapshot();
    });

    test('Should render icon correctly when no source', () => {
      const { toJSON } = setup(
        { uri: null, priority: 'normal' },
        'roundAvatar'
      );
      expect(toJSON()).toMatchSnapshot();
    });
  });
});
