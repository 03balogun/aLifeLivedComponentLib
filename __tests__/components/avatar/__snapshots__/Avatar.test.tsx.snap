// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components >> avatar  Round avatar Should render icon correctly when no source 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#67B6A8",
        "borderRadius": 96,
        "borderWidth": 0,
        "display": "flex",
        "justifyContent": "center",
        "padding": 16,
      },
      Array [
        Object {},
      ],
    ]
  }
/>
`;

exports[`components >> avatar  Round avatar Should render round avatar correctly 1`] = `
<View
  style={
    Array [
      Object {
        "overflow": "hidden",
      },
      Array [
        Object {
          "borderRadius": 96,
          "height": 50,
          "width": 50,
        },
      ],
    ]
  }
>
  <FastImageView
    resizeMode="center"
    source={
      Object {
        "priority": "normal",
        "uri": "https://picsum.photos/id/237/200/300",
      }
    }
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
</View>
`;
